src/App.vue

```diff
@@ -1036,58 +1083,61 @@ export default Vue.extend({
             /* FITRUE_isPython */
         },
         
-        setStateFromPythonFile(completeSource: string, fileName: string, lastSaveDate: number, fileLocation?: FileSystemFileHandle) : void {
-            const allLines = completeSource.split(/\r?\n/);
-            // Split can make an extra blank line at the end which we don't want:
-            if (allLines.length > 0 && allLines[allLines.length - 1] === "") {
-                allLines.pop();
-            }
-            const s = splitLinesToSections(allLines);
-            // Bit awkward but we first copy each to check for errors because
-            // if there are any errors we don't want to paste any:
-            const err = copyFramesFromParsedPython(s.imports.join("\n"), STRYPE_LOCATION.IMPORTS_SECTION, s.importsMapping)
-                        ?? copyFramesFromParsedPython(s.defs.join("\n"), STRYPE_LOCATION.FUNCDEF_SECTION, s.defsMapping)
-                        ?? copyFramesFromParsedPython(s.main.join("\n"), STRYPE_LOCATION.MAIN_CODE_SECTION, s.mainMapping);
-            if (err != null) {
-                const msg = cloneDeep(MessageDefinitions.InvalidPythonParseImport);
-                const msgObj = msg.message as FormattedMessage;
-                msgObj.args[FormattedMessageArgKeyValuePlaceholders.error.key] = msgObj.args.errorMsg.replace(FormattedMessageArgKeyValuePlaceholders.error.placeholderName, err);
-                
-                useStore().showMessage(msg, 10000);
-            }
-            else {
-                // Clear the current existing code (i.e. frames) of the editor
-                this.appStore.clearAllFrames();
-
-                copyFramesFromParsedPython(s.imports.join("\n"), STRYPE_LOCATION.IMPORTS_SECTION);
-                if (useStore().copiedSelectionFrameIds.length > 0) {
-                    getCaretContainerComponent(getFrameComponent(-1) as InstanceType<typeof FrameContainer>).doPaste(true);
+        setStateFromPythonFile(completeSource: string, fileName: string, lastSaveDate: number, fileLocation?: FileSystemFileHandle) : Promise<void> {
```

Comment:
It's a bit hard to tell from the diff because of the indent change, but out of interest, why did this become a promise?